File: 2.js
function fib() {
    var l = [];
    return function() {
        if (l.length == 0) {
            l.unshift(1);
        } else if (l.length == 1) {
            l.unshift(2);
        } else {
            l.unshift(l[0] + l[1]);
        }
        return l[0];
    }
}

var fn = fib();

var n, t = 0;

while ((n = fn()) < 4000000) {
    t += (n % 2 == 0) ? n : 0;
}

console.log(t);

File: 2.ml
open Printf

let sum a b = 
    a + b
    ;;

let rec fib l limit =
    match List.length l with
          0 -> fib (1 :: l) limit
        | 1 -> fib (2 :: l) limit
        | _ -> if ((List.nth l 0) + (List.nth l 1)) > limit then l else fib (((List.nth l 0) + (List.nth l 1)) :: l) limit
    ;;

printf "%d\n" (List.fold_left sum 0 (List.filter (fun n -> n mod 2 == 0) (fib [] 4000000)))
